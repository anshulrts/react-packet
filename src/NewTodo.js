import React, { Component } from 'react';
import './NewTodo.css';

class NewTodo extends Component {
    constructor(props) {
        super(props);
        this.state = { item: '' };
        this.handleUpdate = this.handleUpdate.bind(this);
        // This is the local addTodo
        this.addTodo = this.addTodo.bind(this);
    }
    render() { 
        return ( 
            <div className="NewTodo">
                <input type="text" onChange={ this.handleUpdate } value={ this.state.item } /> &nbsp; &nbsp;
                <button onClick={this.addTodo}>Add</button>
            </div>
        );
    }

    handleUpdate(event) {
        this.setState({ item: event.target.value });
    }

    addTodo() {
        this.props.addTodo(this.state.item);
        this.setState({item : ''});
    }
}
 
export default NewTodo;