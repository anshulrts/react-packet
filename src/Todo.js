import React, { Component } from 'react';
import './Todo.css';

class Todo extends Component {
    render() {
        return <div className="Todo">{ this.props.description }</div>;
    }
}

export default Todo;