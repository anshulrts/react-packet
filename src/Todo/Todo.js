import React, { Component } from 'react';
import styles from  './Todo.module.scss';

class Todo extends Component {

    constructor(props) {
        super(props);
        this.state = {
            description: this.props.description,
            done: false,
            critical: false
        }

        this.markAsDone = this.markAsDone.bind(this);
        this.removeTodo = this.removeTodo.bind(this);
        this.markCritical = this.markCritical.bind(this);
    }

    markAsDone() {
        this.setState({ done: true });
    }

    cssClasses() {
        let classes = [];

        if(this.state.critical) {
            classes = [styles.critical];
        }
        else {
            classes = [styles.todo];
        }
        if(this.state.done) {
            classes = [...classes, styles.done];
        }
        return classes.join(' ');
    }

    removeTodo() {
        this.props.removeTodo(this.state.description);
    }

    markCritical() {
        this.setState({ critical: true });
    }

    render() {
        return (
            <div className={this.cssClasses()}>
                { this.state.description }
            <br />
            <hr className={styles.redDivider} />
            <button className="MarkDone" onClick={ this.markAsDone }>Mark as Done</button>
            <button className="RemoveTodo" onClick={ this.removeTodo }>Remove Me</button>
            <button className="MarkCritical" onClick={ this.markCritical }>Mark as Critical</button>
            </div>
        );
    }
}

export default Todo;