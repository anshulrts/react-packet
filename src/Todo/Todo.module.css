.todo {
    border: 2px solid black;
    text-align: center;
    background: #f5f5f5;
    color: #333333;
    margin: 20px;
    padding: 20px;
}

.done {
    background: #f58888;
  }

  .redDivider {
    border: 2px solid red;
  }