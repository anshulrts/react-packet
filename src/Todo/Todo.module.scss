@import "../shared";

.todo {
    border: 2px solid black;
    text-align: center;
    background: $todo-normal;
    color: #333333;
    margin: 20px;
    padding: 20px;
}

.critical {
    composes: todo;
    background: $todo-critical;
}

.done {
    background: $todo-complete;
  }

  .redDivider {
    border: 2px solid red;
  }