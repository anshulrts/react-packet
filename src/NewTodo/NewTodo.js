import React, { Component } from 'react';
import './NewTodo.css';
import { Button, Input, InputGroup } from "reactstrap";

class NewTodo extends Component {
    constructor(props) {
        super(props);
        this.state = { item: '' };
        this.handleUpdate = this.handleUpdate.bind(this);
        // This is the local addTodo
        this.addTodo = this.addTodo.bind(this);
    }
    render() { 
        return ( 
            <div className="NewTodo">
                <InputGroup>
                    <Input
                        type="text"
                        onChange = { this.handleUpdate }
                        value = { this.state.item }
                        placeholder = "Input item name goes here..."
                    />
                    <Button
                        onClick = { this.addTodo }
                        color = "primary"
                    >
                        Add
                    </Button>
                </InputGroup>
            </div>
        );
    }

    handleUpdate(event) {
        this.setState({ item: event.target.value });
    }

    addTodo() {
        this.props.addTodo(this.state.item);
        this.setState({item : ''});
    }
}
 
export default NewTodo;