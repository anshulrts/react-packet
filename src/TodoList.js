import React, { Component } from 'react';
import Todo from './Todo';
import './TodoList.css';

class TodoList extends Component {
    constructor(props) {
        super(props);
        this.state = { items: ['item #1', 'item #2', 'Item #3'] };
    }

    render() { 
        return ( 
            <div className="TodoList">
                { this.renderItems() }
            </div>
        );
    }

    renderItems() {
        return this.state.items.map(description => (
            <Todo key={description} description={description} />
        ));
    }
}

export default TodoList;
