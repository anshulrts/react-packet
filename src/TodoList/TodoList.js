import React, { Component } from 'react';
import Todo from '../Todo';
import './TodoList.css';
import NewTodo from '../NewTodo';
import Divider from '../Divider';

class TodoList extends Component {
    constructor(props) {
        super(props);
        this.state = { items: ['item #1', 'item #2', 'Item #3'] };
        this.addTodo = this.addTodo.bind(this);
        this.removeTodo = this.removeTodo.bind(this);
    }

    render() { 
        return ( 
            <div className="TodoList">
                <NewTodo addTodo={this.addTodo} />
                { this.renderItems() }
            </div>
        );
    }

    renderItems() {
        return this.state.items.map(description => (
            <div key={"div-" + description}>
                <Todo
                    key={description}
                    description={description}
                    removeTodo={this.removeTodo}
                />
                <Divider key={"divide-" + description} />
            </div>
        ));
    }

    addTodo(item) {
        this.setState({ items: [...this.state.items, item] })
    }

    removeTodo(removeItem) {
        const filteredItems = this.state.items.filter(description => description !== removeItem);
        this.setState({items: filteredItems});
    }
}

export default TodoList;
