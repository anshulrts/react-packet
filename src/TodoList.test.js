import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoList from './TodoList';
import NewTodo from './NewTodo';
import  Todo from './Todo';

describe(TodoList, () => {
    const component = shallow(<TodoList />);

    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<TodoList />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders and matches our snapshot", () => {
        const component = renderer.create(<TodoList />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders a Todo List component", () => {
        expect(component.contains(<div className="TodoList" />));
    });
});