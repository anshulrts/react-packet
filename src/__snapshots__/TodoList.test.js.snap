// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders and matches our snapshot 1`] = `
<div
  className="TodoList"
>
  <div
    className="NewTodo"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
        
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div>
    <div
      className="Todo"
    >
      item #1
      <br />
      <button
        className="MarkDone"
        onClick={[Function]}
      >
        Mark as Done
      </button>
      <button
        className="RemoveTodo"
        onClick={[Function]}
      >
        Remove Me
      </button>
    </div>
    <div
      className="Divider"
    >
      <hr />
    </div>
  </div>
  <div>
    <div
      className="Todo"
    >
      item #2
      <br />
      <button
        className="MarkDone"
        onClick={[Function]}
      >
        Mark as Done
      </button>
      <button
        className="RemoveTodo"
        onClick={[Function]}
      >
        Remove Me
      </button>
    </div>
    <div
      className="Divider"
    >
      <hr />
    </div>
  </div>
  <div>
    <div
      className="Todo"
    >
      Item #3
      <br />
      <button
        className="MarkDone"
        onClick={[Function]}
      >
        Mark as Done
      </button>
      <button
        className="RemoveTodo"
        onClick={[Function]}
      >
        Remove Me
      </button>
    </div>
    <div
      className="Divider"
    >
      <hr />
    </div>
  </div>
</div>
`;
